scripts/add-at-macro/src/add_at_macro/core.clj

```diff
@@ -0,0 +1,374 @@
+          (when required-namespaces
+            (reset! required-namespaces-g required-namespaces))
+          (recur (-> loc (z/replace (z/node current-loc)) z/next)))
+
+        (find-re-com-in-require loc {:macros? true})
+        (do
+          (when verbose?
+            (println "Found re-com.core in :require-macros, checking if `at` macro is referred."))
+          (recur (-> loc (remove-at-from-required-macros verbose?) z/next)))
+
+        :else
+        (recur (z/next loc))))))
+
+(defn check-source-added-already?
+  "Checks if :src option already exists in a vector (re-com component)"
+  [loc]
+  (loop [loc (z/down loc)]
+    (cond
+      (nil? loc) false
+      (-> loc z/string (= ":src")) true
+      :else
+      (recur (z/right loc)))))
+
+(defn add-at-in-component
+  "Adds :src option in a vector, targeting re-com components."
+  [loc verbose?]
+  (if-not (check-source-added-already? loc)
+    (do
+      (when verbose?
+        (println "Adding :src option at line " (first (z/position loc)) " column " (second (z/position loc))))
+      (-> loc z/down
+          (z/insert-right (z/node (z/of-string "(at)")))
+          (z/insert-right (z/node (z/of-string ":src")))
+          z/up))
+    loc))     ;; :src option is already added
+
+(defn find-recom-usages
+  "Find usages of recom namespaces in the document body (Non header section)"
+  [loc parsed-require {:keys [namespaced? verbose? arguments]}]
+  (let [loc                 (if namespaced? (-> loc z/right) loc)  ;; skip the (ns ...) form
+        required-namespaces (if arguments
+                              (clj-set/difference (set (:required-namespaces parsed-require))
+                                                  (set arguments))
+                              (set (:required-namespaces parsed-require)))]
+    (loop [loc loc]
+      (cond
+        (z/end? loc) (z/root loc)
+
+        ;; when re-com component ns is loaded with :refer option
+        (and (z/vector? loc) (seq (z/down loc))
+             (some required-namespaces (conj [] (z/sexpr (z/down loc)))))
+        (recur (-> loc (add-at-in-component verbose?) z/next))
+
+        ;; when re-com component ns is loaded with :as option
+        (and (z/vector? loc) (seq (z/down loc))
+             (-> loc z/down z/string (clojure.string/starts-with? (str (:used-alias parsed-require) "/"))))
+        (recur (-> loc (add-at-in-component verbose?) z/next))
+
+        :else
+        (recur (z/next loc))))))
+
+(defn require-form?
+  "Takes a loc and determines if the file has a namespace form (ns ...)"
+  [loc]
+  (when (-> loc z/down z/string (= "ns"))
+    loc))
+
+(defn a-function?
+  "Takes a loc and determines if the code is a function."
+  [loc]
+  (and (z/list? loc)
+       (-> loc z/down z/string (= "defn"))))
+
+(defn arguments
+  "Takes a function and returns the arguments in the function, the arguments are important to prevent
+   rename when the same name is used for a re-com namespace and a local variable."
```

Comment:
To clarify this docstring, it might be a good idea to provide an example. 

If I understand correctly, you are trying to avoid where an arg is "shadowing" a re-com component. 

```
(defn blah 
   [label arg2 agr3]      ;; <--- oops label is now shadowing a re-com component within this defn
   ...)
   ```

Reply:
Yes, the arguments are to be used to avoid renaming arguments that are shadowing re-com components. 
I am adding an example to the code to clarify further.